terraform/test/main_test.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2015-2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"os/user"
+	"testing"
+	"time"
+
+	"github.com/gravitational/teleport-plugins/lib"
+	"github.com/gravitational/teleport-plugins/lib/logger"
+	"github.com/gravitational/teleport-plugins/lib/testing/integration"
+	"github.com/gravitational/teleport-plugins/terraform/provider"
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type TerraformSuite struct {
+	integration.AuthSetup
+	// client represents plugin client
+	client *integration.Client
+	// teleportConfig represents Teleport configuration
+	teleportConfig lib.TeleportConfig
+	// teleportFeatures represents enabled Teleport feature flags
+	teleportFeatures *proto.Features
+	// plugin represents plugin user name
+	plugin string
+	// terraformConfig represents Terraform provider configuration
+	terraformConfig string
+	// terraformProvider represents an instance of a Terraform provider
+	terraformProvider *schema.Provider
+	// terraformProviders represents an array of providers
+	terraformProviders map[string]func() (*schema.Provider, error)
+}
+
+func TestTerraform(t *testing.T) { suite.Run(t, &TerraformSuite{}) }
+
+func (s *TerraformSuite) SetupSuite() {
+	var err error
+	t := s.T()
+
+	logger.Init()
```

Comment:
The logger setup will be clobbered by the calls to `s.AuthSetup.Setup...()` below. It _might_ be better to pull it out of both here and `AuthSetup`, and explicitly configure the logger in the `TestMain()` for each package.

Reply:
Yes, these lines are artefacts of moving this setup to lib. I just missed their removal. I agree, that might be better to setup logger in TestMain(), but for now, this setup is the same for all plugins. Let me remove these lines and for now keep it like this for the sake of uniformity. We could get back to it in the separate PR.